module/applications/actor/trait-selector.mjs

```diff
@@ -154,7 +175,7 @@ export default class TraitSelector extends BaseConfigSheet {
     const path = Trait.actorKeyPath(this.trait);
     const data = foundry.utils.getProperty(this.document, path);
 
-    this._prepareChoices("choices", `${path}.value`, formData);
+    this._prepareChoices("choices", Trait.changeKeyPath(this.trait), formData);
```

Comment:
Seems like this another potential bug that `${path}.value` was used there instead of `Trait.changeKeyPath(this.trait)` as above.